components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/IdPSecretsProcessor.java

```diff
@@ -70,26 +71,28 @@ public IdentityProvider associateSecrets(IdentityProvider identityProvider) thro
     }
 
     @Override
-    public IdentityProvider addOrUpdateSecrets(IdentityProvider identityProvider) throws SecretManagementException {
+    public IdentityProvider encryptAssociatedSecrets(IdentityProvider identityProvider) throws SecretManagementException {
 
         IdentityProvider clonedIdP;
         try {
             clonedIdP = (IdentityProvider) identityProvider.clone();
             for (FederatedAuthenticatorConfig fedAuthConfig : clonedIdP.getFederatedAuthenticatorConfigs()) {
                 for (Property prop : fedAuthConfig.getProperties()) {
-                    if (prop.isConfidential()) {
-                        String secretName = buildSecretName(clonedIdP.getId(), fedAuthConfig.getName(), prop.getName());
-                        if (secretManager.isSecretExist(IDN_SECRET_TYPE_IDP_SECRETS, secretName)) {
-                            // Update existing secret property.
-                            updateExistingSecretProperty(secretName, prop);
-                            prop.setValue(buildSecretReference(secretName));
-                        } else {
-                            // Add secret to the DB.
-                            if (!StringUtils.isEmpty(prop.getValue())) {
-                                addNewIdpSecretProperty(secretName, prop);
-                                prop.setValue(buildSecretReference(secretName));
-                            }
+                    if (!prop.isConfidential()) {
+                        continue;
+                    }
+                    String secretName = buildSecretName(clonedIdP.getId(), fedAuthConfig.getName(), prop.getName());
+                    if (secretManager.isSecretExist(IDN_SECRET_TYPE_IDP_SECRETS, secretName)) {
+                        // Update existing secret property.
+                        updateExistingSecretProperty(secretName, prop);
+                        prop.setValue(buildSecretReference(secretName));
+                    } else {
+                        // Add secret to the DB.
+                        if (StringUtils.isEmpty(prop.getValue())) {
+                            continue;
```

Comment:
While creating an IDP in carbon console, it will create the federated authenticator and it's properties and store them in the DB. Among those properties, the secret properties won't have any value. Because of that, we are not creating a reference in the secret table as well. 
When it comes to the update operation, there will be a reference of the secret in the IDP_AUTHENTICATOR_PROPERTY. If the secret value is changed as an empty string, the corresponding encrypted value will be replaced in the IDN_SECRET table.